Source/RapyutaSimulationPlugins/Private/Robots/RRBaseRobot.cpp

```diff
@@ -584,61 +584,72 @@ void ARRBaseRobot::Tick(float DeltaSeconds)
     }
 }
 
-void ARRBaseRobot::InitPropertiesFromJSON()
+bool ARRBaseRobot::InitPropertiesFromJSON()
 {
+    // Verify [ROSSpawnParameters], which house JSON
+    if (nullptr == ROSSpawnParameters)
+    {
+        // ELSE: This could only be a BP robot & Editor operations might be being done
+        RR_VERIFY_STATIC_BP_ROBOT(this);
+        return false;
+    }
+    return true;
+
+#if RAPYUTA_SIM_DEBUG

```

Comment:
Isn't this annoying since you will always get log like `[bool_value] not found in json config` with RAPYUTA_SIM_DEBUG==true?

Reply:
Yeah since commenting out codes should be discouraged, and by convention we put it into debug directive to make it formally debugged code, but in this case, it is more of annotation so I suppose better move to doxygen

Reply:
Yeah since commenting out codes should be discouraged, and by convention, we put it into debug directive to be formally denoted as debug code, but in this case, it is more of annotation so I suppose better move to doxygen

Reply:
then please add this
https://www.doxygen.nl/manual/commands.html#cmdcode